### DISCUSSION TITLE
Is result of last command saved in a var?

### DISCUSSION THREAD

**Discussion by yerlaser** [nushell/nushell#6191] [Q&A] [answered]
Probably I missed it in the documentation somewhere but is there a way to reference the results of the last command?
My use case is as follows:
Suppose I just refined a filter to find the exact files that I need and ran that on a large repo with many thousands of files.
Obviously the command takes quite some time to run.
Now I need to access the nth row in the resulting table.
If I rerun the command adding `| get n` the query will take a lot of time again.
How can avoid this?

P.S. I understand that I should've saved the table using `let`, but suppose I forgot to do that.

**Comment by fdncred** [comment, marked as answer]
There is no LAST_COMMAND_RESULTS or anything like that at the moment. What I usually do is `let data = (some | big | pipeline)` and then I access the results over and over again with `$data`.

**Reply by yerlaser** [reply]
OK, fair enough, thanks for the answer!

Would be good though if you could consider such a thing for a future release.
Could save a lot of time for many users.
And, also, I think it will fit to nu's **functional** style better.

**Reply by fdncred** [reply]
We've talked about it but considering that nushell supports infinite streams it makes it tough.

**Reply by wdv4758h** [reply]
Is it equally hard if it's an interactive shell only thing? Like people can use `_` in Python shell to get last result.

**Reply by fdncred** [reply]
It could be done but it would have to have some type of limit, because like I said above, nushell supports infinite streams of data.

**Comment by Deebster** [comment]
Old question, but for future searchers, this is solved as of [v0.115.0](https://www.nushell.sh/blog/2026-08-15-nushell_v0_115_0.html#added-ans-for-accessing-the-last-repl-result).

After you set $env.config.max_last_result_size to a size (e.g. 128MB), $ans.last will contain the last nushell result.
